client/api/src/backend.rs

```diff
@@ -430,6 +430,11 @@ pub trait StorageProvider<Block: BlockT, B: Backend<Block>> {
 		child_info: &ChildInfo,
 		key: &StorageKey,
 	) -> sp_blockchain::Result<Option<Block::Hash>>;
+
+	/// Given a `BlockId` check trie state of all key.
+	/// Return number on unmigrated key for the top state and number
+	/// of unmigrated keys for all child states.
+	fn state_migration_status(&self, id: &BlockId<Block>) -> sp_blockchain::Result<(u64, u64)>;
```

Comment:
Can we not get entirely get rid off this? I don't want to have this function here.

Use something like `state_at().as_trie_backend().storage()` and then you can implement your logic to check the migration status.